File: ruby/checkout/setup_browser_key.rb
require 'ultracart_api'
require_relative '../constants'

# This is a checkout API method. It creates a browser key for use in a client-side checkout.
# This call must be made server-side with a Simple API Key or an OAuth access token.

checkout_api = UltracartClient::CheckoutApi.new_using_api_key(Constants::API_KEY)

key_request = UltracartClient::CheckoutSetupBrowserKeyRequest.new
key_request.allowed_referrers = ["https://www.mywebsite.com"]

browser_key = checkout_api.setup_browser_key(key_request).browser_key

puts browser_key.inspect

File: typescript/checkout/SetupBrowserKey.ts
import {checkoutApi} from '../api';
import {
    CheckoutSetupBrowserKeyRequest,
    CheckoutSetupBrowserKeyResponse
} from 'ultracart_rest_api_v2_typescript';

export class SetupBrowserKey {
    /**
     * Creates a browser key for use in a client-side checkout.
     *
     * This call must be made server-side with a Simple API Key or an OAuth access token.
     */
    public static async execute(): Promise<void> {
        try {
            // Prepare the browser key request
            const keyRequest: CheckoutSetupBrowserKeyRequest = {
                allowed_referrers: ["https://www.mywebsite.com"]
            };

            // Setup the browser key
            const apiResponse: CheckoutSetupBrowserKeyResponse = await checkoutApi.setupBrowserKey({browserKeyRequest: keyRequest});
            const browserKey: string = apiResponse.browser_key || "";

            // Output the browser key
            console.log(browserKey);

        } catch (error) {
            console.error("Error setting up browser key:", error);
        }
    }
}